File: Leetcode/Daily/06-2025/Python3/14.py
# Problem: 2566. Maximum Difference by Remapping a Digit (Easy)
# Link: https://leetcode.com/problems/maximum-difference-by-remapping-a-digit

class Solution:
    def minMaxDifference(self, num: int) -> int:
        
        s = str(num)
        n = len(s)

        maxStr = s
        
        i = 0
        while i < n and s[i] == '9':
            i += 1
        if i < n:
            change = s[i]
            maxStr = s.replace(change, '9')

        change = s[0]
        minStr = s.replace(change, '0')

        return int(maxStr) - int(minStr)

File: Leetcode/Daily/06-2025/JavaScript/14.js
// Problem: 2566. Maximum Difference by Remapping a Digit (Easy)
// Link: https://leetcode.com/problems/maximum-difference-by-remapping-a-digit

/**
 * @param {number} num
 * @return {number}
 */
var minMaxDifference = function(num) {
    let str = num.toString();
    let n = str.length;
    let i = 0;

    let maxStr = str;
    while (i < n && str[i] == '9') i++;
    if (i != n) {
        let change = str[i];
        maxStr = str.split('').map(ch => ch === change ? '9' : ch).join('');
    }
    let max = parseInt(maxStr);

    let minStr = str.split('').map(ch => ch === str[0] ? '0' : ch).join('');
    let min = parseInt(minStr);

    return max -min;
};
